packages/fiori/cypress/specs/ShellBar.cy.ts

```diff
@@ -0,0 +1,235 @@
+		<ui5-avatar slot="profile">
+			<img src="https://sdk.openui5.org/test-resources/sap/f/images/Woman_avatar_01.png" />
+		</ui5-avatar>
+
+		<img slot="logo" src="https://upload.wikimedia.org/wikipedia/commons/5/59/SAP_2011_logo.svg"/>
+
+		<ui5-button icon="nav-back" slot="startButton" id="start-button"></ui5-button>
+
+		<ui5-shellbar-item id="disc" icon="activities" text="Disconnect"></ui5-shellbar-item>
+		<ui5-shellbar-item id="call" icon="sys-help" text="Incoming Calls"></ui5-shellbar-item>
+
+		<ui5-input placeholder="Instructions" slot="searchField" show-suggestions value-state="Information">
+			<div slot="valueStateMessage">Instructions</div>
+		</ui5-input>
+	</ui5-shellbar>`;
+
+	const templateWithOnlyOneAction = html`<ui5-shellbar
+					id="shellbar"
+					primary-title="Product Title"
+					secondary-title="Solution name"
+					notifications-count="99+"
+					show-notifications
+					show-product-switch
+			>
+				<img slot="logo" src="https://upload.wikimedia.org/wikipedia/commons/5/59/SAP_2011_logo.svg"/>
+			</ui5-shellbar>`;
+	beforeEach(() => {
+		cy.mount(basicTemplate).as("html");
+
+		cy.get("#shellbar")
+			.as("shellbar");
+	});
+
+	afterEach(() => {
+		cy.viewport(1920, 1080);
+	});
+
+	it("tests XXL Breakpoint 1920px", () => {
+		cy.viewport(1920, 1080).then(() => {
+			cy.get("@shellbar")
+				.find("ui5-button[slot='startButton']")
+				.as("backButton");
+
+			cy.get("@shellbar")
+				.shadow()
+				.find(".ui5-shellbar-title")
+				.as("primaryTitle");
+
+			cy.get("@shellbar")
+				.shadow()
+				.find(".ui5-shellbar-secondary-title")
+				.as("secondaryTitle");
+
+			cy.get("@shellbar")
+				.shadow()
+				.find(".ui5-shellbar-search-button")
+				.as("searchButton");
+
+			cy.get("@shellbar")
+				.shadow()
+				.find(".ui5-shellbar-custom-item")
+				.as("customActionIcon1");
+
+			cy.get("@shellbar")
+				.shadow()
+				.find(".ui5-shellbar-bell-button")
+				.as("notificationsIcon");
+
+			cy.get("@shellbar")
+				.shadow()
+				.find(".ui5-shellbar-image-button")
+				.as("profileIcon");
+
+			cy.get("@shellbar")
+				.shadow()
+				.find(".ui5-shellbar-button-product-switch")
+				.as("productSwitchIcon");
+
+			cy.get("@shellbar")
+				.find("ui5-toggle-button[slot='assistant']")
+				.as("assistant");
+
+			cy.get("@backButton").should("be.visible");
+			cy.get("@primaryTitle").should("be.visible");
+			cy.get("@secondaryTitle").should("be.visible");
+			cy.get("@customActionIcon1").should("be.visible");
+			cy.get("@notificationsIcon").should("be.visible");
+			cy.get("@profileIcon").should("be.visible");
+			cy.get("@productSwitchIcon").should("be.visible");
+			cy.get("@assistant").should("be.visible");
+
+			cy.get("@searchButton").click();
+
+			cy.get("@shellbar")
+				.find("ui5-input[slot='searchField']")
+				.as("searchField").should("be.visible");
+
+			cy.get("@searchButton").click();
+		});
+	});
+
+	it("tests S Breakpoint 320px", () => {
+		cy.get("html").viewport("iphone-x").then(() => {
```

Comment:
.then is not needed when we switch viewport.

also keep separation of actions. each action to be cy..... 

and put 1 function call per line